Source/BusinessMomentsAdapters/Backend/domain/miners/Miner.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { AggregateRoot, aggregateRoot } from '@dolittle/sdk.aggregates';
+import { Guid } from '@dolittle/rudiments';
+import { EntityAssociatedWithMiner, ImperativeEmbedDefined } from '../../events/miners';
+
+@aggregateRoot('f4d99676-cd37-4e7e-8a03-95cd29b5927d')
+export class Miner extends AggregateRoot {
```

Comment:
I have a question about how we name aggregates.

I think Miner is a good name, but looking on it from the outside it isn't clear to me that this is an Aggregate. Type info about the base class gives me a hint, and when reading the implementation of the class it's super clear to me that this is an aggregate. However, I need to check type info or read the implementation to see that this is an aggregate.

I like to use fuzzy search to find files. If this was named MinerAggregate it would be easy for me to find the aggregates in our solution. For the command handlers for the Adapter we use this convention of naming - see `AdapterCommandHandlers.ts`. 

@einari

Reply:
I realize I'm very inconsistent here.

The thinking is that in the context of working on the domain, you wouldn't by accident use a read model that would happen to have the same name. The thinking has been that structure and context should be enough to know what you're working on. We have principles on naming described here: https://dolittle.io/docs/contributing/guidelines/naming/#prefix--postfix

When searching for instance with ctrl/cmd+P in VSCode you'd see the context and structure it is part of:
<img width="747" alt="image" src="https://user-images.githubusercontent.com/134365/115984017-1bd40a00-a5a5-11eb-8dca-1652ac977c42.png">

But I guess if you're searching for which aggregate roots you have available, that won't help you :) 

Suggest we have a chat about this - the codebase is extremely inconsistent on it - and in fact, most of the constructs postfixes and completely violates the principle :)
Question then would be; would it make sense to strip the postfixes and have clearer names or go the other way and be consistent on postfixes. 

When we put the principle in place, one of the thinkings where that with a pre/post fix you create a conceptual coupling and it is a bit of a "cop out" for naming things properly. But there are nuances to this.

In my head, I don't think about the technical artifact, but what it is doing rather. For instance commands & events, a command through its name represents its intent clearly, not that it is a command - and same for an event, it represents what has happened - not that it is a technical thing of "Event".

Reply:
There could be one benefit to using postfixes; cohesion. 

Right now, we have a structure of `./domain` `./read`in the root of the project and then feature folders within these.
If we postfix `Miner` with `MinerAggregate` or prefix it with `AggregateOfMiner` and then the `MinerReadModel`, we could keep all artifacts related to the same feature for the tier (backend) in the same feature folder and not split out on concern like now. #FoodForThought